client/state/initial-state.js

```diff
@@ -201,7 +184,16 @@ export function persistOnChange( reduxStore ) {
 		window.addEventListener( 'beforeunload', throttledSaveState.flush );
 	}
 
-	reduxStore.subscribe( throttledSaveState );
+	const unsubscribe = reduxStore.subscribe( throttledSaveState );
+
+	return () => {
+		if ( typeof window !== 'undefined' ) {
+			window.removeEventListener( 'beforeunload', throttledSaveState.flush );
+		}
+
+		unsubscribe();
+		throttledSaveState.cancel();
```

Comment:
Is this something that's worth an extra unit test?

Reply:
Yes, added in 2c97a39 🙂